### PR TITLE
BXC 2202: EditLabel action

### PR DESCRIPTION
**Pull request by harringj**

### CODE REVIEW

File: persistence/src/main/java/edu/unc/lib/dl/persist/services/edit/EditLabelService.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        try (Timer.Context context = timer.time()) {
+            aclService.assertHasAccess(
+                    "User does not have permissions to edit labels",
+                    pid, agent.getPrincipals(), Permission.editDescription);
+
+            RepositoryObject obj = repoObjLoader.getRepositoryObject(pid);
+            Model objModel = obj.getModel();
+            Resource resc = objModel.getResource(obj.getUri().toString());
+
+            String oldLabel = replaceOldLabel(objModel, resc, label);
```
[the remark this answers is not in the document]
Comment (harringj) [reply persistence/src/main/java/edu/unc/lib/dl/persist/services/edit/EditLabelService.java]: but createExclusiveRelationship() doesn't let me capture the value of the oldLabel. the main reason i left this method in place is that the requirements state that the following should be part of the premis reporting: ```detail information: 'Object renamed from "<old dc:title>" to "<new label>"'``` so i'm using the method to get the oldLabel. you're right, though, that i don't need it to remove the oldLabel anymore, so maybe i'll just rename the method

File: services/src/main/java/edu/unc/lib/dl/cdr/services/rest/modify/EditLabelController.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+        try {
+            service.editLabel(AgentPrincipals.createFromThread(), pid, label);
+        } catch (Exception e) {
+            result.put("error", e.getMessage());
+            Throwable t = e.getCause();
+            if (t instanceof AuthorizationException || t instanceof AccessRestrictionException) {
+                return new ResponseEntity<>(result, HttpStatus.FORBIDDEN);
+            } else {
+                log.error("Failed to edit label for {} due to {}", pid, e);
```
[the remark this answers is not in the document]
Comment (harringj) [reply services/src/main/java/edu/unc/lib/dl/cdr/services/rest/modify/EditLabelController.java]: ok, i was just following the same pattern we had used in other controllers, but i went ahead and changed it

File: services/src/test/java/edu/unc/lib/dl/cdr/services/rest/modify/EditLabelIT.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    @Test
+    public void testReplaceLabel() throws UnsupportedOperationException, Exception {
+        PID pid = makePid();
+        String oldLabel = "old_work_label";
+        Model workModel = ModelFactory.createDefaultModel();
+        workModel.add(workModel.createResource(pid.getRepositoryPath()), DcElements.title,
+                oldLabel);
+        WorkObject work = repositoryObjectFactory.createWorkObject(pid, null);
```
[the remark this answers is not in the document]
Comment (harringj) [reply services/src/test/java/edu/unc/lib/dl/cdr/services/rest/modify/EditLabelIT.java]: yes, i did. thanks for catching that!

File: services/src/test/resources/edit-label-it-servlet.xml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        http://www.springframework.org/schema/context 
+        http://www.springframework.org/schema/context/spring-context-3.0.xsd
+        http://www.springframework.org/schema/mvc
+        http://www.springframework.org/schema/mvc/spring-mvc-3.0.xsd">
+        
+    <mvc:annotation-driven/>
+
+    <context:component-scan resource-pattern="**/EditLabelController*" base-package="edu.unc.lib.dl.cdr.services.rest.modify"/>
+    
+    <bean id="addContainerService" class="edu.unc.lib.dl.persist.services.edit.EditLabelService">
```
[the remark this answers is not in the document]
Comment (harringj) [reply services/src/test/resources/edit-label-it-servlet.xml]: yeah, i was having copy pasta for lunch yesterday. mmm!

File: persistence/src/main/java/edu/unc/lib/dl/persist/services/edit/EditLabelService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (bbpennel) [reply persistence/src/main/java/edu/unc/lib/dl/persist/services/edit/EditLabelService.java]: Good point, you're totally right that we still need to get the old value. I'll mention that there is a shortcut for getting the resource, `obj.getResource()` if you'd like to save some steps

File: persistence/src/main/java/edu/unc/lib/dl/persist/services/edit/EditLabelService.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+            repoObjFactory.createExclusiveRelationship(obj, DcElements.title, label);
+
+            obj.getPremisLog()
+            .buildEvent(Premis.Migration)
```
[the remark this answers is not in the document]
Comment (bbpennel) [reply persistence/src/main/java/edu/unc/lib/dl/persist/services/edit/EditLabelService.java]: looks the same :(

File: persistence/src/main/java/edu/unc/lib/dl/persist/services/edit/EditLabelService.java [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    public void setOperationsMessageSender(OperationsMessageSender operationsMessageSender) {
+        this.operationsMessageSender = operationsMessageSender;
+    }
+
+    private String getOldLabel(RepositoryObject obj) {
+        String oldLabel = "no 'dc:title'";
+        Model objModel = obj.getModel();
+        Resource resc = obj.getResource();
+        if (objModel.contains(resc, DcElements.title)) {
+        Statement s = objModel.getRequiredProperty(resc, DcElements.title);
```
Comment (bbpennel) [inline persistence/src/main/java/edu/unc/lib/dl/persist/services/edit/EditLabelService.java]: indent

**Pull request merged**